### PR TITLE
Fix ValidateUpdate on the validating webhook

### PR DESCRIPTION
**Pull request by tiraboschi**
Fix ValidateUpdate on the validating webhook

Fix ValidateUpdate on the validating webhook
properling looking up for existing objects
correctly initializing their names.

Testing it on e2e CI

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by ovirt-infra** [comment]
All tests passed

File: hack/deploy.sh [hunk 1]
```diff
@@ -153,6 +153,8 @@ done
 # compare initial cluster SCCs to be sure HCO deployment didn't introduce any change
 dump_sccs_after
 
+"${CMD}" patch hyperconverged -n ${HCO_NAMESPACE} ${HCO_RESOURCE_NAME} --type='json' -p='[{"op": "add", "path": "/spec/infra/nodePlacement/nodeSelector/nodeType", "value": "infra" }, {"op": "add", "path": "/spec/workloads/nodePlacement/nodeSelector/nodeType", "value": "workloads" }]'
```
Comment (nunnatsa) [inline hack/deploy.sh:156]: This is not working. I tested this version and it worked for me:
```
"${CMD}" patch hyperconverged -n ${HCO_NAMESPACE} ${HCO_RESOURCE_NAME} --type='json' -p='[{"op": "add", "path":"/spec/infra", "value": {"nodePlacement": {"nodeSelector": {"nodeType": "infra"}}}}, {"op": "add", "path": "/spec/workloads", "value": {"nodePlacement": {"nodeSelector": {"nodeType": "workloads"}}}}]'
```

File: hack/upgrade-test.sh [hunk 2]
```diff
@@ -245,4 +245,8 @@ ${CMD} get deployments -n ${HCO_NAMESPACE} -o yaml | grep image | grep -v imageP
 ${CMD} get pod $HCO_CATALOGSOURCE_POD -n ${HCO_CATALOG_NAMESPACE} -o yaml | grep image | grep -v imagePullPolicy
 
 dump_sccs_after
+
+Msg "Test node placement APIs"
+"${CMD}" patch hyperconverged -n ${HCO_NAMESPACE} ${HCO_RESOURCE_NAME} --type='json' -p='[{"op": "add", "path": "/spec/infra/nodePlacement/nodeSelector/nodeType", "value": "infra" }, {"op": "add", "path": "/spec/workloads/nodePlacement/nodeSelector/nodeType", "value": "workloads" }]'
```
Comment (nunnatsa) [inline hack/upgrade-test.sh:250]: Please change to:
```
"${CMD}" patch hyperconverged -n ${HCO_NAMESPACE} ${HCO_RESOURCE_NAME} --type='json' -p='[{"op": "add", "path":"/spec/infra", "value": {"nodePlacement": {"nodeSelector": {"nodeType": "infra"}}}}, {"op": "add", "path": "/spec/workloads", "value": {"nodePlacement": {"nodeSelector": {"nodeType": "workloads"}}}}]'
```

**Review by nunnatsa — CHANGES_REQUESTED** [review]
The `kubectl patch` command is not working. Suggested a working version.

File: hack/deploy.sh (hunk 1 above)
Comment (tiraboschi) [reply hack/deploy.sh]: Done, thanks

File: hack/upgrade-test.sh (hunk 2 above)
Comment (tiraboschi) [reply hack/upgrade-test.sh]: Done, thanks

**Review by nunnatsa — APPROVED** [review]

**Comment by ovirt-infra** [comment]
All tests passed

**Comment by tiraboschi** [comment]
done on #847

**Pull request closed without merging**